frontend/src/features/admin/agencies/EditAgencyPage.tsx

```diff
@@ -0,0 +1,215 @@
+
+  const checkAgencies = (
+    <Select
+      label="Parent Agency"
+      field="parentId"
+      data-testid="agency"
+      options={selectAgencies}
+      placeholder={agency?.parentId ? undefined : 'Please select if applicable'}
+    />
+  );
+
+  const goBack = () => {
+    history.push('/admin/agencies');
+  };
+
+  const initialValues: IAgencyDetail = {
+    parentId: agency.parentId,
+    code: agency.code,
+    id: agency.id,
+    name: agency.name,
+    description: agency.description,
+    isDisabled: agency.isDisabled,
+    sendEmail: agency.sendEmail,
+    email: agency.email,
+    rowVersion: agency.rowVersion,
+  };
+
+  const newValues: any = {
+    code: '',
+    name: '',
+    description: '',
+    isDisabled: false,
+    sendEmail: true,
+    email: '',
+    rowVersion: '',
+  };
+
+  return (
+    <div>
+      {showDelete && <DeleteModal {...{ showDelete, setShowDelete, history, dispatch, agency }} />}
+      <Navbar className="navBar" expand="sm" variant="light" bg="light">
+        <Navbar.Brand href="#">
+          {' '}
+          <FaArrowAltCircleLeft onClick={goBack} size={20} /> Agency Information
+        </Navbar.Brand>
+      </Navbar>
+      <Container fluid={true}>
+        <Row className="agency-edit-form-container">
+          <Formik
+            enableReinitialize
+            initialValues={newAgency ? newValues : initialValues}
+            onSubmit={async (values, { setSubmitting }) => {
+              if (!newAgency) {
+                dispatch(
+                  getUpdateAgencyAction(
+                    { id: agencyId },
+                    {
+                      id: agency.id,
+                      name: values.name,
+                      code: values.code,
+                      email: values.email,
+                      isDisabled: values.isDisabled,
+                      sendEmail: values.sendEmail,
+                      parentId: Number(values.parentId),
+                      description: values.description,
+                      rowVersion: values.rowVersion,
+                    },
+                  ),
+                );
+              } else {
+                const data = await createAgency({
+                  name: values.name,
+                  code: values.code,
+                  email: values.email,
+                  isDisabled: values.isDisabled,
+                  sendEmail: values.sendEmail,
+                  parentId: Number(values.parentId),
+                  description: values.description,
+                })(dispatch);
+                dispatch(clear(actionTypes.ADD_PARCEL));
+                history.replace(`/admin/agency/${data.id}`);
+              }
+              setSubmitting(false);
+            }}
+          >
+            {props => (
+              <Form className="agencyInfo">
+                <Label>Agency</Label>
+                <Input data-testid="agency" field="name" value={props.values.name} type="text" />
```

Comment:
Could use the label prop here instead potentially.
If not (possibly due to styling issues) would prefer you use label for instead of data-testid